### ISSUE TITLE
Add support for inspecting variables

### ISSUE THREAD

**Issue by sbrunk** [almond-sh/almond#360]
https://github.com/lckr/jupyterlab-variableInspector is a JupyterLab extension that shows currently used variables and their values. If we're able to access to access the variables in scope somehow in Ammonite, it should be possible to support this in almond.

**Comment by alexarchambault** [comment]
I've been having that in mind for a few days too (and @jvican just alluded the subject to me via [this post](https://devblogs.microsoft.com/python/python-in-visual-studio-code-april-2019-release/).

From what I remember of https://github.com/lckr/jupyterlab-variableInspector, we should need some minimal tweaking in it (so that it gets enabled for scala), then on the kernel side, we need to expose some method calls listing (as display data) a JSON that the variable inspector can understand.

I guess the trickier should be to keep track of what variables get defined. That may require tweaking the [preprocessor](https://github.com/almond-sh/almond/blob/753818b02a7f9408ee5a2f9094fe07ca20225a32/modules/scala/scala-interpreter/src/main/scala/almond/AmmInterpreter.scala#L121) (that can adjust the user code a bit before it's compiled).

**Comment by sbrunk** [comment]
Yeah too bad the vscode Jupyter support is still limited to python. :(

Seems like the minimal tweaking needs to be done in [variableinspector.ts](https://github.com/lckr/jupyterlab-variableInspector/blob/7a284978a2654bf937078ece8172da453f65f816/src/inspectorscripts.ts#L4-L8) by providing a `LanguageModel ` for Scala and adding it to `scripts`.

```typescript
type LanguageModel = {
  initScript: string;
  queryCommand: string;
  matrixQueryCommand: string;
}
```

As for the server side, the default REPL exposes things like
```
scala> $intp.definedSymbolList
```
I was hoping that Ammonite had something similar built we could use to avoid having to track variables ourselves, but my knowledge of Ammonite is still somewhat limited.
